src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.SingleSegment.Tests.cs

```diff
@@ -0,0 +1,132 @@
+            s_multipleEmptyMultiLineComment = Encoding.UTF8.GetBytes("/**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**//**/{}");
+            s_singleShortMultiLineComment = Encoding.UTF8.GetBytes("/*asdasd*/{}");
+            s_multipleShortMultiLineComment = Encoding.UTF8.GetBytes("/*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*//*asdasd*/{}");
+            s_singleLongMultiLineComment1 = Encoding.UTF8.GetBytes("/*" + new string('c', 2000) + "*/{}");
+            s_singleLongMultiLineComment2 = Encoding.UTF8.GetBytes("/*" + new string('\n', 2000) + "*/{}");
+
+            s_singleEmptySingleLineComment = Encoding.UTF8.GetBytes("//\n{}");
+            s_multipleEmptySingleLineComment = Encoding.UTF8.GetBytes("//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n//\n{}");
+            s_singleShortSingleLineComment = Encoding.UTF8.GetBytes("//asdasd\n{}");
+            s_multipleShortSingleLineComment = Encoding.UTF8.GetBytes("//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n//asdasd\n{}");
+            s_singleLongSingleLineComment = Encoding.UTF8.GetBytes("//" + new string('c', 2000) + "\n{}");
+
+            MultiSegmentSetup();
+        }
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleEmptyMultiLineComment_CommentHandling_Allow() => GenericReadTest(s_singleEmptyMultiLineComment, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultipleEmptyMultiLineComment_CommentHandling_Allow() => GenericReadTest(s_multipleEmptyMultiLineComment, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleShortMultiLineComment_CommentHandling_Allow() => GenericReadTest(s_singleShortMultiLineComment, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultipleShortMultiLineComment_CommentHandling_Allow() => GenericReadTest(s_multipleShortMultiLineComment, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleLongMultiLineComment1_CommentHandling_Allow() => GenericReadTest(s_singleLongMultiLineComment1, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleLongMultiLineComment2_CommentHandling_Allow() => GenericReadTest(s_singleLongMultiLineComment2, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void SingleEmptySingleLineComment_CommentHandling_Allow() => GenericReadTest(s_singleEmptySingleLineComment, JsonCommentHandling.Allow);
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void MultipleEmptySingleLineComment_CommentHandling_Allow() => GenericReadTest(s_multipleEmptySingleLineComment, JsonCommentHandling.Allow);
```

Comment:
The test case name is capturing too many concepts that it is hard to parse. I would have trouble reasoning about the results as well.

Let's break it down `MultipleEmptySingleLineComment_CommentHandling_Allow`:
- Multiple - This is the count of how many comments show up (Single vs Multiple) - Maybe count params?
- Empty - So, empty is the talking about the contents of the comment itself, vs short/long.
- SingleLine - OK, so this is multi-line comment vs single line (/**/ vs //\n), which I think we can live without OR add a - - bool params to say singleline true/false)
- Allow - this can be fixed with a CommentHandling params

We should figure out a cleaner way to represent such test cases.